CMSgov/onemac
# OY2-17027 - Create SPA package forms

Story: https://qmacbis.atlassian.net/browse/OY2-17027
Endpoint: https://dtkk4o21ssg53.cloudfront.net/

### Details

Restructure OneMacForm and existing form types (Base Waiver, Temp extension) and creation of Medicaid SPA and CHIP SPA forms for package view.

### Changes

- created form types for each of (Base Waiver, Temp Extension, Medicaid Spa, Chip Spa) and reused properties from common/type to enforce uniform declaration/usage
- removed "uploads" properties from common/type definitions
- refactored OneMACForm.tsx to utilize new properties
- added proposed effective date as property on Medicaid SPA and CHIP SPA

### Test Plan

1. Login as state submitter
2. Create Medicaid SPA and CHIP SPA
3. Ensure Proposed Effective date is a required field
